sdk/keyvault/azure-security-test-keyvault-jca/src/test/java/com/azure/security/keyvault/jca/ServerSocketTest.java

```diff
@@ -147,79 +176,116 @@ public void testServerSocket() throws Exception {
     }
 
     /**
-     * Test SSLServerSocket WITH self-signed client trust.
+     * Test SSLServerSocket with default trust manager.
      *
      * @throws Exception when a serious error occurs.
      */
     @Test
-    public void testServerSocketWithSelfSignedClientTrust() throws Exception {
+    public void testServerSocketWithDefaultTrustManager() throws Exception {
 
         /*
-         * Add JCA provider.
+         * Setup server side.
+         *
+         *  - Set the SSL context to use the KeyManagerFactory.
+         *  - Create the SSLServerSocket using th SSL context.
          */
-        KeyVaultJcaProvider provider = new KeyVaultJcaProvider();
-        Security.addProvider(provider);
+
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        tmf.init(ks);
+
+        SSLContext context = SSLContext.getInstance("TLS");
+        context.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
+
+        SSLServerSocketFactory factory = context.getServerSocketFactory();
+        SSLServerSocket serverSocket = (SSLServerSocket) factory.createServerSocket(8768);
+        serverSocket.setNeedClientAuth(true);
+
+        startSocket(serverSocket);
+
+        /*
+         * Setup client side
+         *
+         * - Create an SSL context.
+         * - Set SSL context to trust any certificate.
+         */
+
+        SSLContext sslContext = SSLContexts
+            .custom()
+            .loadTrustMaterial(ks, new TrustSelfSignedStrategy())
+            .loadKeyMaterial(ks, "".toCharArray(), new ClientPrivateKeyStrategy())
+            .build();
+
+        /*
+         * And now execute the test.
+         */
+        String result = sendRequest(sslContext, "8768");
+
+        /*
+         * And verify all went well.
+         */
+        assertEquals("Success", result);
+    }
+
+
+    /**
+     * Test SSLServerSocket with key vault trust manager.
+     *
+     * @throws Exception when a serious error occurs.
+     */
+    @Test
+    public void testServerSocketWithKeyVaultTrustManager() throws Exception {
+
+        Security.insertProviderAt(new KeyVaultTrustManagerFactoryProvider(), 1);
```

Comment:
The difference between `testServerSocketWithDefaultTrustManager` and `testServerSocketWithKeyVaultTrustManager` are:
1. `Security.insertProviderAt(new KeyVaultTrustManagerFactoryProvider(), 1);`.
2. port.

Could you please reuse other codes?